tests/servers/tests.py

```diff
@@ -57,29 +56,57 @@ def test_protocol(self):
         with self.urlopen('/example_view/') as f:
             self.assertEqual(f.version, 11)
 
-    @override_settings(MIDDLEWARE=[])
     def test_closes_connection_without_content_length(self):
         """
-        The server doesn't support keep-alive because Python's http.server
-        module that it uses hangs if a Content-Length header isn't set (for
-        example, if CommonMiddleware isn't enabled or if the response is a
-        StreamingHttpResponse) (#28440 / https://bugs.python.org/issue31076).
+        A HTTP 1.1 server is supposed to support keep-alive. Since our
+        development server is rather simple we support it only in cases where
+        we can detect a content length from the response. This should be doable
+        for all simple views and streaming responses where an iterable with
+        length of one passed.
```

Comment:
This comes directly from wsgiref: https://github.com/python/cpython/blob/master/Lib/wsgiref/handlers.py#L191 -- I'll add a comment. If the iterable has only one item, one can immediatelly evaluate it, since it will be in memory anyways.